resources/assets/js/components/FindSkillsDialog/FindSkillsDialog.tsx

```
@@ -0,0 +1,435 @@
+      elementSelector: "#skill-search-button",
+    };
+    prevTabbedElement.current = document.getElementById("skill-search-button");
+    return Promise.resolve();
+  };
+
+  /**
+   * Sets up the state of the skill results pane when a skill category is clicked.
+   * @param childSkillCategory Skill Category object.
+   */
+  const onSkillCategoryClick = (childSkillCategory: SkillCategory): void => {
+    setFirstVisit(false);
+    searchResultsSource.current = {
+      elementSelector: `#${childSkillCategory.key}-skill-category`,
+    };
+    setActiveCategory(childSkillCategory.key);
+    setResultsSectionText({
+      title: intl.formatMessage(messages.skills, {
+        category: localizeFieldNonNull(locale, childSkillCategory, "name"),
+      }),
+      description: "",
+      // TODO: Restore this when discriptions are added to Skill Categories on backend.
+      // localizeFieldNonNull(
+      //   locale,
+      //   childSkillCategory,
+      //   "description",
+      // ),
+    });
+    setResults(categoryIdToSkillsMap.get(childSkillCategory.id) ?? []);
+  };
+
+  /**
+   * Resets the search results pane to its default state, and puts focus back on the search results source.
+   */
+  const resetResults = (): void => {
+    setFirstVisit(true);
+    focusOnElement(searchResultsSource.current.elementSelector);
+    searchResultsSource.current = { elementSelector: "" };
+    setActiveCategory("");
+    setResults([]);
+  };
+
+  const dialogRef = React.useRef<HTMLElement | null>(null);
+  const handleTabableElements: (
+    event: KeyboardEvent,
+  ) => void = React.useCallback(
+    (event) => {
+      if (event.key === "Tab" && dialogRef.current) {
+        const tabableElements = Array.from(
+          dialogRef.current.querySelectorAll(
+            "[data-tabable='true']:not([disabled])",
+          ),
+        ) as HTMLElement[];
+
+        if (tabableElements.length === 0) {
+          event.preventDefault(); // TODO: should this throw an error?
```

Let's decide properly what to do, and get rid of that TODO.
I'm thinking now that preventDefault is probably the worst solution - it stops the tab key from working, and it does so silently 😆 . I'm thinking we should either throw an error ("A valid dialog must contain at least one focusable element") or we return _without_ preventing default, which allows the browser to handle the tab behavior. @yonikid15 what do you think.

---

Well technically a dialog should **always** have at least a close button, so it shouldn't ever enter this block. I'm fine with with removing the prevent default, but then we should add in a close button in the top right corner (ie. x icon) that's permanently on every dialog.

---

Can we guarantee that, with the way we allow end users to build the contents of the modal out of sub-components?

---

We can permanently add it to the `<Header>` sub component, leave it trailing after the children.